app/routes.js

```diff
@@ -8,23 +8,32 @@ router.get('/', async (req, res) => {
 });
 
 router.get('/component/:component', async (req, res) => {
-  const { component } = req.params;
-  generateTemplate({ component });
+  const name = req.params.component;
+  generateTemplate({ name }, 'component');
 
-  res.render(`templates/${component}-template`);
+  res.render(`templates/components/${name}-template`);
 });
 
 router.post('/component/:component', async (req, res) => {
-  const { component } = req.params;
-  const formParams = req.body;
-  generateTemplate({ component, formParams });
+  const name = req.params.component;
+  const formParams = JSON.parse(req.body.params);
+  generateTemplate({ name, formParams }, 'component');
```

Comment:
please change to `generateTemplate({ name, formParams, templateType: 'component' });`